jinahub/indexers/SimpleIndexer/__init__.py

```diff
@@ -80,7 +81,7 @@ def search(self, docs: 'DocumentArray', parameters: Dict, **kwargs):
         for _q, _ids, _dists in zip(flat_docs, idx, dist):
             for _id, _dist in zip(_ids, _dists):
                 d = Document(self._docs[int(_id)], copy=True)
-                d.scores['cosine'] = 1 - _dist
+                d.scores[self.distance_metric] = 1 - _dist
```

Comment:
This does not make much sense for `euclidean` distance then?

Also let's use `match` interface from `DocumentArray`

Reply:
Ah true, thanks for pointing that out. I have updated the search function to use `docs.match`. Since we want the score/similarity in this case, what do you think of using `lambda: q_emb, d_emb: -self.distance(q_emb, d_emb)` as the distance metric?

Reply:
The negative sign is now removed from the metric function. I have also doubled check places where `SimpleIndexer` is used to ensure the usage is correct after the modification.